### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by mandelatuks**
#### [the link to the video documentation](https://www.loom.com/share/9ca8b62a3c9042b68cb9e945b0a4e42e)

- Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

- The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing

### CODE REVIEW

File: app/views/splash/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<%= link_to 'Sign up', new_user_registration_path, class: 'sign-in-btn'%>
+</header>
+
+<section class="splash-page">
+<h1> Welcome To Money Wallet </h1>
+<p>The Best Way To Manage Your Money</p>
+<div class= "splash-actions">
+<%= link_to 'Log-in', user_session_path, class: 'sign-in-btn' %>
+</div>
+</section>
```
Comment (AnselemOdims) [inline app/views/splash/index.html.erb]: - [ ] Please kindly ensure that your design matches the original design guidelines. At the moment, the buttons in the splash screen, login and signup pages do not match the design.

### Your Design
image

### Original Design
image

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up", class: 'transaction-btn'%>
+  </div>
```
Comment (AnselemOdims) [inline app/views/devise/registrations/new.html.erb]: - [ ] Please kindly ensure that a user is able to register to the app with their full name included. This should be mandatory to register

File: app/views/transactions/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+<div id="transactions">
+  <% @transactions.each do |transaction| %>
+    <%= render transaction %>
+  <% end %>
+</div>
+
+<footer>
+<%= link_to "Add Transaction", new_category_transaction_path %>
+</footer>
```
Comment (AnselemOdims) [inline app/views/transactions/index.html.erb:22]: - [ ] Please kindly ensure that the transactions and categories page matches the original design guidelines. See image below for reference

image

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @mandelatuks , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/transactions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
+class TransactionsController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_transaction, only: %i[show edit update destroy]
+
+  # GET /transactions or /transactions.json
+  def index
```
Comment (rustam-tolipov) [inline app/controllers/transactions_controller.rb:7]: - [x] You made a great effort! However, there is one thing that needs to be completed. I can see only the category and total amount but I cannot seed the transactions. In transactions, you need to provide all transactions as well.

Screen Shot 2022-09-09 at 14 36 21

> Transactions page
> 
> 
> For a given category, the list of transactions is presented ordered by the most recent
> At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category)
> There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction
> When the user clicks on the "Back" button (<), the user navigates to the home page

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello, @mandelatuks, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/transactions_controller.rb (hunk 4 above)
Comment (mandelatuks) [reply app/controllers/transactions_controller.rb:7]: @Rustamxon7 i had implemented it but i guess the previous reviewer was clear about the changes 
Thanks for the review 
Screenshot from 2022-09-09 11-55-15

**Review by rustam-tolipov — APPROVED** [review]
## Hello @mandelatuks 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by mandelatuks** [comment]
@Rustamxon7 
Thank you so much for approving this PR 
happy coding :smile:

**Pull request merged**